src/internal/base-component/__tests__/metrics.test.ts

```diff
@@ -212,25 +213,33 @@ describe('Client Metrics support', () => {
             },
             1
           );
-          checkMetric(`awsui_pkg_d${testCase[1][0]}`, ['main', 'pkg', 'default', 'used', 'react', testCase[1][1]]);
+          checkMetric(`awsui_pkg_d${testCase[1][0]}`, {
+            o: 'main',
+            s: 'pkg',
+            t: 'default',
+            a: 'used',
+            f: 'react',
+            v: testCase[1][1],
+            c: undefined,
+          });
         });
       });
     });
   });
 
   describe('sendMetricObjectOnce', () => {
-    test('logs a metric only once if same source and action', () => {
+    test('logs a metric only once if it is the same object', () => {
       const metricObj = {
         source: 'pkg',
-        action: 'used',
+        action: 'used' as const,
         version: '5.0',
       };
 
       metrics.sendMetricObjectOnce(metricObj, 1);
       metrics.sendMetricObjectOnce(metricObj, 1);
       expect(window.AWSC.Clog.log).toHaveBeenCalledTimes(1);
     });
-    test('logs a metric only once if same source and action but different versions', () => {
+    test('logs metric for each different version if same source and action', () => {
```

Comment:
Strange test here. It asserts that `component@v1` and `component@v2` will be reported only once. We do not have multiple versions in a single bundle, so we do not rely on this behavior everywhere.

However, it blocks reporting `component@variant=1` and `component@variant-2` as separate metrics, so I had to update it here.